pkg/testutil/stdout.go

```diff
@@ -0,0 +1,27 @@
+package testutil
+
+import (
+	"bytes"
+	"sync"
+)
+
+// MockStdout provides a concurrent-safe wrapper around bytes.Buffer to
+// replicate the behavior of os.Stdout (or os.Stderr) without
+// introducing potential data races from the buffer resizing itself
+// during Write calls (see bytes.(*Buffer).tryGrowByReslice()).
```

Comment:
Might be worth adding an extra bit of context to this implementation so it's clear we're referring to the yacc Spinner (as I believe that was what originally prompted this new code).